strawberryfields/decompositions.py

```diff
@@ -400,6 +400,114 @@ def mach_zehnder(m, n, internal_phase, external_phase, nmax):
     return BS @ Rinternal @ BS @ Rexternal
 
 
+def mach_zehnder_inv(m, n, phi_int, phi_ext, nmax):
+    r"""The inverse of the mach_zehnder matrix."""
+    return np.transpose(np.conj(mach_zehnder(m, n, phi_int, phi_ext, nmax)))
+
+
+def nullMZi(m, n, U):
+    r"""Nullifies element m,n of U using mach_zehnder_inv"""
+    (nmax, mmax) = U.shape
+
+    if nmax != mmax:
+        raise ValueError("U must be a square matrix")
+
+    if U[m, n] == 0:
+        phi_i = np.pi
+        phi_e = 0
+    elif U[m, n + 1] == 0:
+        phi_i = 0
+        phi_e = 0
+    else:
```

Comment:
Are these the checks that take care of edge cases like the identity?

Reply:
Yes, exactly! One if for identity-like cases, another one for divide by zero.

Reply:
@shreyapkumar: maybe worth adding short comments to the if-else statement, specifying which branch is for the identity-like case, which one is for the divide by zero case :)